### PR TITLE
Submit stable Voyager chart

### PR DESCRIPTION
**Pull request by saumanbiswas**
cc: @sadlil @TamalSaha

### CODE REVIEW

**Comment by seanknox** [comment]
Hi @saumanbiswas, thanks for your contribution! Can you provide some background on what Voyager is and why it might be relevant in the charts repo?

**Comment by linki** [comment]
@saumanbiswas your Chart isn't structured correctly. `Chart.yaml` etc. should live under `stable/voyager`, not `stable/voyager/voyager`. Can you just move them over?

File: stable/voyager/voyager/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    chart: {{.Chart.Name}}-{{.Chart.Version}}
+    heritage: {{.Release.Service}}
+    release: {{.Release.Name}}
+    run: {{ template "fullname" . }}
```
Comment (linki) [inline stable/voyager/voyager/templates/deployment.yaml]: this label name should rather be `app`

File: stable/voyager/voyager/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Configuration
+
+The following tables lists the configurable parameters of the Voyager chart and their default values.
+
+
+| Parameter                  | Description                  | Default              |
+| -----------------------    | ---------------------------- | -------------------- |
+| `image`                    |  Container image to run      | `appscode/voyager`   |
+| `imageTag`                 |  Image tag of container      | `1.5.1`              |
```
Comment (linki) [inline stable/voyager/voyager/README.md]: please also document `cloudProvider` and `clusterName`, their meaning and default values, if any.

File: stable/voyager/voyager/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  template:
+    metadata:
+      labels:
+        run: {{ template "fullname" . }}
+    spec:
+      containers:
+      - args:
+        - --cloud-provider={{.Values.cloudProvider}}
+        - --cluster-name={{.Values.clusterName}}
+        - --v=3
```
Comment (linki) [inline stable/voyager/voyager/templates/deployment.yaml]: The value of `--v` should be configurable or just leave it at their default?

File: stable/voyager/voyager/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        image: "{{.Values.image}}:{{.Values.imageTag}}"
+        name: {{ template "name" . }}
```
Comment (linki) [inline stable/voyager/voyager/templates/deployment.yaml]: I think it's fine to just call it "voyager".

File: stable/voyager/voyager/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        ports:
+        - containerPort: 1234
+          name: zero
```
Comment (linki) [inline stable/voyager/voyager/templates/deployment.yaml]: Are these things (port 1234, name: zero) something related to voyager? If not I would prefer just using 8080 and name: http here or something similar.

File: stable/voyager/voyager/templates/svc.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    chart: {{.Chart.Name}}-{{.Chart.Version}}
+    heritage: {{.Release.Service}}
+    release: {{.Release.Name}}
+    run: {{ template "fullname" . }}
```
Comment (linki) [inline stable/voyager/voyager/templates/svc.yaml]: This should be `app` as well.

File: stable/voyager/voyager/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  name: {{ template "fullname" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      run: {{ template "fullname" . }}
+  template:
+    metadata:
+      labels:
+        run: {{ template "fullname" . }}
```
Comment (linki) [inline stable/voyager/voyager/templates/deployment.yaml]: This should be `app` as well.

File: stable/voyager/voyager/templates/deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    chart: {{.Chart.Name}}-{{.Chart.Version}}
+    heritage: {{.Release.Service}}
+    release: {{.Release.Name}}
+    run: {{ template "fullname" . }}
+  name: {{ template "fullname" . }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      run: {{ template "fullname" . }}
```
Comment (linki) [inline stable/voyager/voyager/templates/deployment.yaml]: This should be `app` as well.

File: stable/voyager/voyager/templates/svc.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+  name: {{ template "fullname" . }}
+spec:
+  ports:
+  - name: zero
+    port: 1234
+    targetPort: zero
+  selector:
+    run: "{{ template "fullname" . }}"
```
Comment (linki) [inline stable/voyager/voyager/templates/svc.yaml]: This should be `app` as well.

**Review by linki — DISMISSED** [review]
Chart needs to be moved to the correct folder in order to run the tests.

**Comment by viglesiasce** [comment]
Marking as stale. @saumanbiswas please update in the next week if you'd like this to be merged.

File: stable/voyager/values.yaml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+##
+## Voyager chart configuration
+##
+image: appscode/voyager
+imageTag: 1.5.1
```
Comment (tamalsaha) [inline stable/voyager/values.yaml]: @saumanbiswas , use image tag 1.5.5

**Comment by linki** [comment]
@k8s-bot ok to test

File: stable/voyager/templates/NOTES.txt [hunk 11]
```diff
@@ -0,0 +1,5 @@
+Set cloudProvider, clusterName for installing Voyager
+
+To verify that Voyager has started, run:
+
+  kubectl --namespace={{ .Release.Namespace }} get deployments -l "release={{ .Release.Name }}, run={{ template "fullname" . }}"
```
Comment (linki) [inline stable/voyager/templates/NOTES.txt]: This still references the `run` label.

**Comment by linki** [comment]
@seanknox FWIK, voyager is an ingress controller, much like the nginx ingress controller.

File: stable/voyager/templates/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,32 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    chart: {{.Chart.Name}}-{{.Chart.Version}}
+    heritage: {{.Release.Service}}
+    release: {{.Release.Name}}
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/voyager/templates/deployment.yaml]: Please use `{{ template "name" . }}` for the `app` label. Apply everywhere.

File: stable/voyager/templates/svc.yaml [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    release: {{.Release.Name}}
+    app: {{ template "fullname" . }}
+  name: {{ template "fullname" . }}
+spec:
+  ports:
+  - name: http
+    port: 8080
+    targetPort: http
+  selector:
+    app: "{{ template "fullname" . }}"
```
Comment (unguiculus) [inline stable/voyager/templates/svc.yaml]: After changing the `app` as mentioned above, the selector should be as follows:

```yaml
app: {{ template "name" . }}
release: {{ .Release.Name }}
```

File: stable/voyager/templates/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,32 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    chart: {{.Chart.Name}}-{{.Chart.Version}}
```
Comment (unguiculus) [inline stable/voyager/templates/deployment.yaml]: Please use consistent formatting, i. e. space after `{{` and space before `}}`. Apply everywhere.

File: stable/voyager/Chart.yaml [hunk 15]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+description: Voyager provides controller for Ingress and Certificates for Kubernetes developed by AppsCode.
+icon: https://cdn.appscode.com/images/icon/voyager.png
+name: voyager
+version: 0.1.0
+sources:
+  - https://github.com/appscode/voyager
+maintainers:
+  - name: 'AppsCode Inc.'
```
Comment (unguiculus) [inline stable/voyager/Chart.yaml]: Use Github username.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by tamalsaha** [comment]
@unguiculus , I have addressed your comments.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**